eva-accession-pipeline/src/main/resources/monitoring/monitor_duplicate_accessions.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+from datetime import datetime
+import getpass
+import smtplib
+from __init__ import *
+
+
+def export_mongo_accessions(mongo_connection_properties, collection_name, export_output_filename):
+    export_command = 'mongoexport --host {0} --port {1} --db {2} --username {3} --password {4} ' \
+                     '--authenticationDatabase={5} --collection {6} --type=csv --fields accession -o "{7}" ' \
+                     '--sort \'{{accession: 1}}\''\
```

Comment:
I'm wondering if it's better to sort on the server (putting pressure on the server) or to move the sorting to the output file which would run wherever the script is run.

Reply:
In theory this is a [covered query](https://docs.mongodb.com/manual/core/query-optimization/#covered-query), so it should run just scanning the "accession" index, and the indexes are usually kept in memory, so it shouldn't put much pressure. However, I wouldn't put my hand over the fire about it, as MongoDB is surprisingly picky on this stuff.

Reply:
the covered query does run without any issues on PROD and it is covered by the index. If not, mongoexport will fail per specification if the output is more than 100MB - see [here](https://docs.mongodb.com/master/reference/program/mongoexport/#cmdoption-mongoexport-sort)

Reply:
ok